cloudferrylib/os/network/neutron.py

```diff
@@ -106,14 +106,11 @@ def read_info(self, **kwargs):
         # Get full list off busy segmentation IDs
         used_seg_ids = get_segmentation_ids_from_net_list(full_nets_list)
 
-        routers = []
+        routers = self.get_routers()
         subnet_ids = {sn['id'] for sn in subnets}
-        for router in self.get_routers():
-            router_subnet_ids = subnet_ids & set(router['subnet_ids'])
-            if router_subnet_ids:
-                router['subnet_ids'] = list(router_subnet_ids)
-                routers.append(router)
-                LOG.debug('Got router: %s', router['id'])
+        for router in routers:
+            router['subnet_ids'] = [sn_id for sn_id in router['subnet_ids']
```

Comment:
Actual result will be:

    public_notShared_network_t1 <=> router_t1 <=> private_notShared_network_t1
    public_notShared_network_t2 <=> router_t2 <=> private_notShared_network_t2

I will have only 2 routers.